frontend/spec/features/add_as_you_go_spec.rb

```diff
@@ -0,0 +1,379 @@
+# frozen_string_literal: true
+require 'spec_helper'
+require 'rails_helper'
+
+describe 'Add As You Go buttons', js: true do
+  before(:each) do
+    login_admin
+  end
+
+  context 'for non-Agent record types' do
+    shared_examples 'top-level non-Notes subforms' do |type, num_expected_subforms, num_expected_header_buttons|
```

Comment:
The title of the `shared_examples` block should read well when used in an `it_behaves_like` block, maybe something like: "a top level form without Notes subforms" ?